lib/Support/parser.cpp

```diff
@@ -0,0 +1,238 @@
+#include <vector>
+#include <stack>
+#include <queue>
+#include <iostream>
+
+#include "parser.h"
+#include "lexer.h"
+
+using namespace std;
+
+int Parser::hash(Token tt) {
+	TokenType t = tt.token_type;
+	if (t== NOT_TOKEN) {
+		return 0;
+	}
+	else if (t == AND_TOKEN) {
+		return 1;
+	}
+	else if (t == OR_TOKEN) {
+		return 2;
+	}
+	else if (t == LPAREN_TOKEN) {
+		return 3;
+	}
+	else if (t == RPAREN_TOKEN) {
+		return 4;
+	}
+	else if (t == VARIABLE_TOKEN) {
+		return 5;
+	} 
+	else{ //END_TOKEN
+		return 6;
+	}
+}
+
+enum comp {
+	lessThan, greaterThan, equall, error, accept
+};
+comp precedanceTable[7][7];
```

Comment:
This should be initialized in place instead of in the `buildOperationPrecedenceTable` function.

```cpp
static constexpr precendenceTable[7][7] = { ... };
```